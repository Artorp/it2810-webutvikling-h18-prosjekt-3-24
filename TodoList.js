import React from 'react';
import { StyleSheet, Platform, StatusBar } from 'react-native';
import {
  List,
  ListItem,
  Body,
  Button,
  Container,
  Content,
  Footer,
  FooterTab,
  Header,
  Icon,
  Left,
  Right,
  Text,
  Title,
} from 'native-base';

import { Actions } from 'react-native-router-flux';
import TodoListItem from './TodoListItem';
import NewTodoButton from './NewTodoButton';
import StepCounter from './components/StepCounter';

export default class TodoList extends React.Component {
  constructor() {
    super();
    this.items = [];
    for (let i = 0; i < 60; i++) this.items.push('ListeoppfÃ¸ring');
    // console.log("this.items i TodoList:", this.items);
  }

  scrollViewArray() {
    return this.items.map((item, key) => (
      <ListItem key={key}>
        <TodoListItem />
      </ListItem>
    ));
  }

  render() {
    return (
      <Container style={styles.container}>
        <Header>
          <Left>
            <Button transparent>
              <Icon name="menu" />
            </Button>
          </Left>
          <Body>
            <Title>Header</Title>
          </Body>
          <Right />
        </Header>
        <Content padder>
          <StepCounter />
          <NewTodoButton
            onPress={() => {
              Actions.editTodo();
            }}
          />
          <List>{this.scrollViewArray()}</List>
        </Content>
        <Footer>
          <FooterTab>
            <Button full>
              <Text>Fane 1</Text>
            </Button>
          </FooterTab>
          <FooterTab>
            <Button full>
              <Text>Fane 2</Text>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    ...Platform.select({
      android: {
        marginTop: StatusBar.currentHeight,
      },
    }),
  },
});
