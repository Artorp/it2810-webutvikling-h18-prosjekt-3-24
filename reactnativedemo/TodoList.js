import React from "react";
import { StyleSheet, View } from "react-native";
import { List, ListItem, Text } from "native-base";

import TodoListItem from "./TodoListItem";
import NewTodoButton from "./NewTodoButton";

import { Actions } from 'react-native-router-flux';

export default class TodoList extends React.Component {
  constructor() {
    super();
    this.items = [];
    for (i = 0; i < 60; i++) this.items.push("ListeoppfÃ¸ring");
    // console.log("this.items i TodoList:", this.items);
  }

  scrollViewArray() {
    return this.items.map((item, key) => (
      <ListItem key={key}>
        <TodoListItem />
      </ListItem>
    ));
  }

  render() {
    return (
      <View>
        <NewTodoButton
          onPress= {() => {Actions.editTodo(); }}
        />
        <List>{this.scrollViewArray()}</List>
      </View>
    );
  }
}
