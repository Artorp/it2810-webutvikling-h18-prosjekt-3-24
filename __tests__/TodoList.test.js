import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../components/TodoList';

const TodoList2 = class extends TodoList {
  setState(newState) {
    console.log('setState mocket: ' + newState.items);
    this.state.items = newState.items;
  }
};

test('blir tegnet riktig', () => {
  const tree = renderer.create(<TodoList />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('newItem og deleteItem virker', async () => {
  const list = new TodoList2();
  expect(list.state.items).toHaveLength(0);

  const todoName = 'A';
  const editTodoState = {
    chosenDate: new Date(),
    nameOfTodo: todoName,
  };
  list.newItem(editTodoState);
  expect(list.state.items).toHaveLength(1);

  console.log('prøver å fjerne «' + list.state.items[0] + '».');
  await list.deleteItem(list.state.items[0]);
  expect(list.state.items).toHaveLength(0);
});
