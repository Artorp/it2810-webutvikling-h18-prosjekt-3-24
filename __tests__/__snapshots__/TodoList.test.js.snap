// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blir tegnet riktig 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "flex": 1,
      },
    ]
  }
  virtual={undefined}
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
          virtual={undefined}
        >
          Supermotivator 3000
        </Text>
      </View>
    </View>
  </View>
</View>
`;
